catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.kt

```diff
@@ -502,6 +517,155 @@ class DataListFragment : Fragment(),
+            } else {
+                currentSprite.allBricks
+            }
+            if (!checkIfItemIsUsed(item, brickFlatList)) {
+                itemsToDelete.add(item)
+            }
+        }
+        return itemsToDelete
+    }
+
+    private fun checkIfItemIsUsed(item: UserData<*>, brickFlatList: List<Brick>): Boolean {
+        for (brick in brickFlatList) {
+            if (isItemInUse(item, brick)) {
+                showInUseAlert(item)
+                return true
+            }
+        }
+
+        for (brick in brickFlatList) {
+            if (isItemInDropDown(item, brick)) {
+                showInDropDownDialog(item, brickFlatList)
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun isItemInUse(item: UserData<*>, brick: Brick): Boolean {
+        val itemName = item.name
+
+        if (brick is FormulaBrick) {
+            for (formula in brick.formulas) {
+                val formulaTree: FormulaElement = formula.root
+
+                if (item is UserVariable && formulaTree.containsVariable(itemName)) {
+                    return true
+                } else if (item is UserList && formulaTree.containsList(itemName)) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+
+    private fun isItemInDropDown(item: UserData<*>, brick: Brick): Boolean {
+        val itemName = item.name
+
+        if (brick is UserVariableBrickWithFormula && item is UserVariable && isItemNameAndVariableNameSame(itemName, brick)) {
+            return true
+        } else if (brick is UserListBrick && item is UserList && isItemNameAndListNameSame(itemName, brick)) {
+            return true
+        }
+        return false
+    }
+
+    @VisibleForTesting
+    fun deleteUserDataItemFromDropDown(item: UserData<*>, brickFlatList: List<Brick?>) {
+        val itemName = item.name
+
+        for (brick in brickFlatList) {
+            if (brick is UserVariableBrickWithFormula && item is UserVariable && isItemNameAndVariableNameSame(itemName, brick)) {
+                brick.userVariable = null
+            } else if (brick is UserListBrick && item is UserList && isItemNameAndListNameSame(itemName, brick)) {
+                brick.userList = null
+            }
+        }
+        deleteItem(item)
+    }
+
+    private fun getBrickList(sceneList: List<Scene>): List<Brick> {
+        val brickFlatList: List<Brick> = ArrayList()
+
+        for (scene in sceneList) {
+            for (sprite in scene.spriteList) {
+                for (script in sprite.scriptList) {
+                    script.addToFlatList(brickFlatList)
+                }
+            }
+        }
+        return brickFlatList
+    }
+
+    private fun isItemNameAndVariableNameSame(itemName: String, brick: Brick): Boolean =
+        (brick as UserVariableBrickWithFormula).userVariable != null && itemName === brick.userVariable.name
```

Comment:
```suggestion
        (brick as UserVariableBrickWithFormula).userVariable?.name == itemName
```
Can be simplified. 

Also do you need them to be the same object? Because === checks, that the 2 point to the same object. 

Same for the isItenNameAndListNameSame